### PR TITLE
My followers following search engine server

### PR DESCRIPTION
**Pull request by prastoin**
close https://github.com/AdonisEnProvence/MusicRoom/issues/430

### CODE REVIEW

**Review by Devessier — APPROVED** [review]
Great job!
I asked several questions, not crucial.

File: packages/server/app/Controllers/Http/SearchUsersController.ts [hunk 1]
```diff
@@ -14,6 +19,70 @@ import UserService from 'App/Services/UserService';
 
 const SEARCH_USERS_LIMIT = 10;
 
+async function listUserFollowers({
```
[the remark this answers is not in the document]
Comment (prastoin) [reply packages/server/app/Controllers/Http/SearchUsersController.ts:22]: Yes totally
In fact we do have some services, but indeed this not accurate everywhere

File: packages/server/tests/ListMyFollowers.test.ts [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+            .post(
+                urlcat(TEST_MY_PROFILE_ROUTES_GROUP_PREFIX, 'search/followers'),
+            )
+            .send({
+                page: 1,
+                searchQuery: '',
+                tmpAuthUserID: meUserID,
+            } as ListMyFollowersRequestBody)
+            .expect(404);
+    });
```
[the remark this answers is not in the document]
Comment (prastoin) [reply packages/server/tests/ListMyFollowers.test.ts:265]: As the related route is the `me/search/followers` it means that user doesn't have to check his own visibility to be able to retrieve it's data
It's a bit weird as for the moment we pass a `tmpAuthUserID` but in the future he will be computed from the request auth

File: packages/server/tests/ListMyFollowing.test.ts [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+            .post(
+                urlcat(TEST_MY_PROFILE_ROUTES_GROUP_PREFIX, 'search/following'),
+            )
+            .send({
+                page: 1,
+                searchQuery: '',
+                tmpAuthUserID: meUserID,
+            } as ListMyFollowingRequestBody)
+            .expect(404);
+    });
```
[the remark this answers is not in the document]
Comment (prastoin) [reply packages/server/tests/ListMyFollowing.test.ts:265]: As the related route is the `me/search/following` it means that user doesn't have to check his own visibility to be able to retrieve it's data
It's a bit weird as for the moment we pass a `tmpAuthUserID` but in the future he will be computed from the request auth

File: packages/server/tests/ListMyFollowers.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Devessier) [reply packages/server/tests/ListMyFollowers.test.ts:265]: Thank you! I misunderstood the purpose of this PR. 🙏

**Pull request merged**